### PR TITLE
OCPBUGS-2028:  Override kubelet hostname for GCP platform

### PR DESCRIPTION
**Pull request by jrvaldes**
This change fetches the hostname from the GCP instance metadata service
and overrides the kubelet hostname, so that the name of the Windows node
matches the hostname.

### CODE REVIEW

**Comment by mansikulkarni96** [comment]
For my understanding, what is the difference between the default behaviour that truncates the hostname over 63 characters and the code introduced in this PR?

**Comment by jrvaldes** [comment]
> For my understanding, what is the difference between the default behaviour that truncates the hostname over 63 characters and the code introduced in this PR?


@mansikulkarni96, there is no default behavior that truncates the hostname for Windows nodes, it only [exists](https://github.com/openshift/machine-config-operator/blob/e46a003cdb705f30ace3acdfff830ca36446f4e5/templates/common/_base/files/usr-local-bin-mco-hostname.yaml#L25) for Linux instances. This PR introduces this functionality for Windows instances in GCP.

**Comment by jrvaldes** [comment]
/close

As WMCB is getting deprecated, this change should be part of WICD's kubelet service implementation and proposed in a separate PR after the following PRs merge:
- https://github.com/openshift/windows-machine-config-operator/pull/1260
- https://github.com/openshift/windows-machine-config-operator/pull/1279

File: pkg/cloud/gcp.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	"cloud.google.com/go/compute/metadata"
+	"github.com/pkg/errors"
+)
+
+// gcpHostnameMaxLength is the maximum number of character allowed for the instance's hostname in GCP
+const gcpHostnameMaxLength = 63
+
+// processGCPInstanceHostname truncates the given hostname if the length is longer 63 characters, otherwise returns
+// the original value
+func processGCPInstanceHostname(hostname string) string {
```
Comment (aravindhp) [inline pkg/cloud/gcp.go:15]: Is there an existing function in MAO or MCO that you can leverage?

Comment (jrvaldes) [reply pkg/cloud/gcp.go:15]: I looked and couldn't find anything.

**Pull request closed without merging**